File: LeetCode/Python/0026. Remove Duplicates from Sorted Array #2.py
class Solution:
    def removeDuplicates(self, arr: List[int]) -> int:
        last_unique = 0
        n = len(arr)
        
        for i in range(1, n):
            if arr[i] != arr[last_unique]:
                arr[i], arr[last_unique + 1] = arr[last_unique + 1], arr[i]
                last_unique += 1
        
        return last_unique + 1
            

File: LeetCode/Java/0026. Remove Duplicates from Sorted Array #2.java
class Solution {
    public int removeDuplicates(int[] arr) {
        int n = arr.length;
        int lastUnique = 0;
        
        for (int i = 1; i < n; i++) 
            if (arr[i] != arr[lastUnique])
                swap(arr, i, ++lastUnique);
        
        return ++lastUnique;
    }
    
    private void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}
